rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java

```diff
@@ -1507,9 +1507,19 @@ private J.VariableDeclarations visitVariables(List<VariableTree> nodes, Space fm
 
         List<J.Annotation> typeExprAnnotations = collectAnnotations(annotationPosTable);
 
-        TypeTree typeExpr;
-        if (vartype == null || vartype instanceof JCErroneous) {
+        TypeTree typeExpr = null;
+        if (vartype == null) {
             typeExpr = null;
+        } else if (vartype instanceof JCErroneous) {
+            try {
+                if (source.substring(node.startPos, node.pos).startsWith("var")) {
+                    // "var x = unknownType" can be translated into "(ERROR) var x = unknownType"
+                    typeExpr = new J.Identifier(randomId(), sourceBefore("var"), Markers.EMPTY, emptyList(), "var", typeMapping.type(vartype), null);
+                    typeExpr = typeExpr.withMarkers(typeExpr.getMarkers().add(JavaVarKeyword.build()));
+                }
+            } catch (Exception e) {
+                typeExpr = null;
+            }
```

Comment:
@knutwannheden is probably the better judge of any possible adverse effects if we were to adopt this, but I found it also works without the extra conditional block, by falling into the var-handling else block just below.

Should we keep this try-catch or can we rely on the existing logic, effectively only removing the ` || vartype instanceof JCErroneous` above?
```suggestion
```

Reply:
Looks cleaner. Without any adverse effects, let's apply your suggestion.